lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -4082,6 +4110,61 @@ deserializeBabbageTx :: ByteString -> Cardano.Tx Cardano.BabbageEra
 deserializeBabbageTx = either (error . show) id
     . Cardano.deserialiseFromCBOR (Cardano.AsTx Cardano.AsBabbageEra)
 
+-- Ideally merge with 'updateSealedTx'
+addExtraTxIns
+    :: [TxIn]
+    -> Cardano.Tx Cardano.BabbageEra
+    -> Cardano.Tx Cardano.BabbageEra
+addExtraTxIns ins = modifyBabbageTxBody $ \body ->
+    body
+        { Babbage.inputs = Babbage.inputs body
+            <> (Set.map (Cardano.toShelleyTxIn . toCardanoTxIn) (Set.fromList ins))
+        }
+
+withValidityBeyondHorizon
+    :: PartialTx Cardano.BabbageEra
+    -> PartialTx Cardano.BabbageEra
+withValidityBeyondHorizon ptx = ptx
+    { tx = modifyBabbageTxBody setValidityBeyondHorizon (tx ptx)
+    }
+
+  where
+    -- Set validity beyond horizon used in dummyEraHistoryBounded
+    setValidityBeyondHorizon
+        :: Babbage.TxBody StandardBabbage
+        -> Babbage.TxBody StandardBabbage
+    setValidityBeyondHorizon body = body
+        { Babbage.txvldt =
+            ValidityInterval
+                (SJust $ SlotNo 1_000_000)
+                (SJust $ SlotNo 2_000_000)
```

Comment:
Note: these are both far ahead in the future, when compared to the `40` in the `dummyForkInterpreter`.